src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -382,6 +382,51 @@ namespace JSONHelpers
         SaveFancyZonesData();
     }
 
+    bool FancyZonesData::IsAnotherWindowOfApplicationInstanceZoned(HWND window) const
+    {
+        std::scoped_lock lock{ dataLock };
+        auto processPath = get_process_path(window);
+        if (!processPath.empty())
+        {
+            auto history = appZoneHistoryMap.find(processPath);
+            if (history != appZoneHistoryMap.end())
+            {
+                DWORD processId = 0;
+                GetWindowThreadProcessId(window, &processId);
+
+                auto processIdIt = history->second.processIdToHandleMap.find(processId);
+
+                if (processIdIt == history->second.processIdToHandleMap.end())
+                {
+                    return false;
+                }
+                else if (processIdIt->second != window && IsWindow(processIdIt->second))
+                {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    void FancyZonesData::UpdateHandle(HWND window)
```

Comment:
`UpdateHandle` should be renamed to clarify what this function is updating, something like `UpdateProcessIdToHandleMap`